src/method/technology.tex

```diff
@@ -0,0 +1,100 @@
+user interfaces.
+It is developed and maintained by Meta, and is used by many large companies.
+React is focused on mobile applications first and foremost, but the team's focus was on the desktop experience.
+
+% textidote: ignore begin % due to node.js and next.js being identified as an url
+The backend was built using Node.js~\cite{node.js}, which allows the team to build the server using JavaScript.
+Node.js is a popular choice for backend development, as it is fast and scalable.
+As the team is using React for client-side rendering, they decided to also incorporate Next.js~\cite{next.js} to handle
+the server-side rendering.
+Next.js is a React framework running on top of Node.js, which makes it easier to build the frontend and backend
+together.
+% textidote: ignore end
+
+As Stockfish~\cite{stockfish} is built on C++, the team had to find a way to interact with it.
+To do this, the team created a REST API using Python, as it was deemed the easiest to implement using
+py-stockfish~\cite{py-stockfish} as a wrapper.
+The API was used to create functions that would take input from the server, run it through Stockfish, and return the
+output back to the server.
+Unit tests were used to ensure that the API was working correctly.
+
+\subsection{JavaScript libraries}\label{subsec:libraries}
+
+Throughout the project the team used a number of different libraries to aid with the mundane tasks of web development.
+% textidote: ignore begin % Due to chess.js being identified as an url
+For example, as the focus was on the learning aspect of the project, the team used Chess.js~\cite{chess.js} to handle
+the chess logic and react-chessboard~\cite{react-chessboard} to display the chessboard.
+% textidote: ignore end
+This saved a lot of time, as chess is a complex game with many rules.
+
+The team also used Material UI~\cite{mui} (MUI) to style the website, as it is a modern and popular library that
+uses Google's design language, Material Design.
+MUI was used to create the buttons, text fields, and other components that make up the website.
+
+Lastly, the team used Chatscope~\cite{chatscope} to create the chat window of the website.
+The chat window is an important element of the website, as it allows users to communicate with Stockfish or with each
+other.
+
+\subsection{Hosting provider}\label{subsec:hosting}
+
+When it came to hosting the website, the team had to choose between a number of different providers.
+Amazon Web Services~\cite{aws} was considered, but it was deemed too expensive for the team's needs.
+Instead, the team settled down on Firebase~\cite{firebase}, which is a platform developed by Google.
+The main reason behind this decision was that Firebase allowed for easy authentication with Google accounts, which made
+it easier to integrate user accounts into the website.
+Firebase is also free up to a certain point, which suited the team's budget.
```

Comment:
It sounds a little like a novel. In my opinion it reads a little too much like storytelling. Consider changing it to a more straightforward tone.